Provide a flat namespace combining env and system props

jklukas opened the pull request:
Addresses #16 

I largely removed the use of `env.` and `sys.` from the README, assuming that the new non-namespaced style would generally be preferred. @tkrille, if you think that shouldn't be preferred usage, I can revert some of the README changes.

jklukas commented:
Checking in on this. Any thoughts, @tkrille?

tkrille commented:
I'm on it. Looks good on the first view.

tkrille commented:
I would completely remove any evidence of the `env.` and `sys.` namespaces, except for a hint in the README, that using these namespaces is also possible.

tkrille commented:
This also includes the tests, of course :)

tkrille commented:
The implementation looks really good. We should think about what we're doing with the documentation and tests.

jklukas commented:
I just pushed a change to the README that recasts `sys` and `env` as only necessary for complex situations. Also added a link to the specific section of Freemarker docs that discusses what variable names are available from a template.

Pushed an additional commit that removes `sys.` and `env.` prefixes from tests except to explicitly test usage of those variables.

tkrille commented on README.md:
```diff
@@ -189,12 +185,12 @@ any other Freemarker features beyond simple variable interpolation:
     - type: http
-      port: ${env.PORT!8080}
+      port: ${PORT!8080}
 <#if env.ENABLE_SSL == 'true'>
```
there's still an `env.` here.

tkrille commented:
`SystemPropertiesSpec.groovy` is missing the test for the `sys.` namespace.

tkrille commented:
Looks good to me now. Please fix the last 2 comments and squash all commits. Thanks again for your work!

jklukas commented:
Added a `'can use sys prefix'` test and removed the final `env.` reference in the README. Squashed.

Thanks for the review, @tkrille. I think this is ready to merge.

tkrille commented:
Awesome! Thank you very much!

tkrille commented:
When do you need a release of the new features?

jklukas commented:
> When do you need a release of the new features?

I'd love to see a release in the next week or two, but I'm not under any particular time constraints.

tkrille commented:
Alright, there are currently holidays in Germany, so maybe I'll do it next Monday.

jklukas commented:
That would be fine.

|| Jeff Klukas
|| http://jeff.klukas.net

On Sat, May 14, 2016 at 9:11 AM, Thomas Krille <notifications@github.com>
wrote:

> Alright, there are currently holidays in Germany, so maybe I'll do it next
> Monday.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/tkrille/dropwizard-template-config/pull/17#issuecomment-219219656>
>
